service/async/default_server.go

```diff
@@ -28,16 +30,71 @@ type defaultSever struct {
 	svc        Service
 }
 
-func NewDefaultAPIServerWithGin(
+func NewDefaultAsyncServersWithGin(
 	rootCtx context.Context,
 	cfg config.Config,
 	processStore persistence.ProcessStore,
 	visibilityStore persistence.VisibilityStore,
 	logger log.Logger,
-) Server {
+) []Server {
+	var servers []Server
+	addressToServerMap := map[string]Server{}
+
+	serverAddresses := strings.Split(cfg.AsyncService.ClientAddress, ",")
```

Comment:
`cfg.AsyncService.ClientAddress` is the address for api service to call async service. API service needs to notify the async serivce's specific node/shard. But api service doesn't need to use memberlist to lookup. Instead, api service will send to a random node, and then the receiving async service node may forward the request to the right node of the shard.

Because api service doesn't need to know which node own the shard, it can send request behind a LBS.

Reply:
The current way is:
1. api server send the internal request to a random async server
2. the random async server forward the request to the target async server 

This is the same as what you described above. See the codes near the comment `// randomly send the request to an async service`.